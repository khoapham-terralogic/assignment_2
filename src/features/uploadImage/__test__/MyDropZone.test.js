import React from 'react'
import { shallow } from 'enzyme';
import MyDropzone from '../MyDropZone';

describe('Test <MyDropzone />', () => {
    let props = {
        parentCallback: jest.fn(),
        isActive: false
    }
    it('should render', () => {
        const wrapper = shallow(<MyDropzone {...props} />)
        expect(wrapper).toMatchSnapshot()
    });
    it('should render correspondingly to isActive props', () => {
        const wrapper = shallow(<MyDropzone {...props} />)
        wrapper.setProps({ isActive: true })
        expect(wrapper.find(".dropzone--active__p").text()).toEqual("Your file has been selected, choose another?")
    })
    it('should render correspondingly to isDragActive props', () => {
        const wrapper = shallow(<MyDropzone isDragActive={false} {...props} />)
        wrapper.setProps({ isDragActive: true })
        expect(wrapper.find('.dropzone--inactive__p').text()).toEqual("Drag 'n' drop some files here, or click to select files")
    })
})
