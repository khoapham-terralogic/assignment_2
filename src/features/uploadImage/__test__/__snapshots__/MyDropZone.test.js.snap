// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <MyDropzone /> should render 1`] = `
<div
  className="dropzone dropzone--inactive"
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <input
    autoComplete="off"
    multiple={true}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />
  <img
    alt="upload"
    className="dropzone__img"
    src="upload.svg"
  />
  <p
    className="dropzone--inactive__p"
  >
    Drag 'n' drop some files here, or click to select files
  </p>
</div>
`;
